cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/AbstractDataprocProvisioner.java

```diff
@@ -90,6 +95,21 @@ public final ProvisionerSpecification getSpec() {
   @Override
   public final void initialize(ProvisionerSystemContext systemContext) {
     this.systemContext = systemContext;
+    try {
+      if (Files.exists(DataprocUtils.CACHE_DIR_PATH)) {
+        DataprocUtils.deleteDirectoryContents(DataprocUtils.CACHE_DIR_PATH.toFile());
+      }
+      File cacheDir = Files.createDirectory(DataprocUtils.CACHE_DIR_PATH).toFile();
+      createTwillAndLauncherJar(cacheDir);
+    } catch (Exception e) {
+      LOG.warn("Unable to cache twill and launcher jar", e);
```

Comment:
We should differentiate between exceptions that might happen when invalidating the cache, and when generating the new cache. 
I believe we need to retry if cache invalidation fails, and after multiple retry, fail the whole provisioner if we are not sure that cache has been invalidated.

Reply:
We are caching the jars in `/tmp` which always gets cleared on restart. Wondering do we even need to worry about restart??

Reply:
Removed the cache invalidation logic and added a comment for future reference.

Reply:
Discussed offline, to do only cache invalidation in initialize with retries and fail the provisioner ultimately.